doc/src/learn/cryptography/sui-move-signatures.md

```diff
@@ -0,0 +1,167 @@
+---
+title: Sui On-Chain Signatures Verification in Move
+---
+
+Move contracts in Sui supports verifications for several signature schemes on chain. Note that not all signatures supported in on-chain verification are supported as user signature verification. See [Sui Signatures](sui-signatures.md#user-signature) for valid signatures to be used for transaction execution. 
+
+This topic covers:
+ 1. How to use [fastcrypto](https://github.com/MystenLabs/fastcrypto)'s CLI tool to create a signature of a given scheme. For testing and debugging only, do not use in production.
+ 1. Call the Move method on-chain to verification by submitting the signature, the message and the public key. 
+
+Signature schemes covered: 
+ 1. Ed25519 signature (64 bytes)
+ 1. Secp256k1 non-recoverable signature (64 bytes)
+ 1. Secp256k1 recoverable signature (65 bytes)
+ 1. Secp256r1 non-recoverable signature (64 bytes)
+ 1. Secp256r1 recoverable signature (65 bytes)
+ 1. BLS G1 signature (minSig setting)
+ 1. BLS G2 signature (minPk setting)
+
+## Usage
+
+1. Set up fastcrypto CLI binary. 
+
+```shell
+git@github.com:MystenLabs/fastcrypto.git
+cd fastcrypto/
+cargo build --bin sigs-cli
+```
+
+2. Sign with CLI and submit to on-chain Move method.
+
+ 1. Ed25519 signature (64 bytes)
+```shell
+target/debug/sigs-cli keygen --scheme ed25519 --seed 0000000000000000000000000000000000000000000000000000000000000000                
+Private key in hex: $SK
+Public key in hex: $PK
+
+target/debug/sigs-cli sign --scheme ed25519 --msg $MSG --secret-key  $SK
+
+Signature in hex: $SIG
+Public key in hex: $PK
+```
+
+Call smart contract's verify method. All inputs are represented in bytes in hex format:
```

Comment:
is this part of step 3 or a new step? It is hard to tell with the current formatting. Is the 1. in the section above actually step 3, or a new step 1? It renders as 3

Reply:
i edited it to make it clearer. 

there are two subpoints under `1. Ed25519 signature (64 bytes)`: `* Generate a key and sign a message. ` and `* Call the verify method in Move.`

same for all signature schemes, feel free to edit if its still not clear.